### PR TITLE
Astro 1256 select update

### PR DESCRIPTION
**Pull request by Cheshire89**
https://rocketcom.atlassian.net/browse/ASTRO-1256

renamed dropdown to select
update select styling
removed redundant variables

### CODE REVIEW

**Review by cskerbo — COMMENTED** [review]
@dmcalester Is there a desire or need to update the Readme and various links to reference select instead of drop-down?

Screen Shot 2021-02-15 at 9 22 57 AM

**Comment by amalyah** [comment]
@Cheshire89 Could you add the link to the relevant Jira ticket in the description of this branch? Helps me track time better.

**Review by amalyah — CHANGES_REQUESTED** [review]
- Bottom of the Select menu does not need to have the sharp bottom edges when the field is not expanded. Could we return the radii to the same as the top corners?
- In general, is it possible to use color variables instead of color declarations throughout the CSS files?

File: static/css/astro.core.css [hunk 1]
```diff
@@ -1848,37 +1835,36 @@ label {
...
-
-  border: 1px solid var(--selectBorderColor);
-  border-radius: 3px;
-  border-radius: var(--buttonBorderRadius);
-
-  color: var(--selectTextColor);
+  border: 1px solid var(--inputBorderColorAlt);
+  border-radius: 3px 3px 0 0;
+  border-radius: var(--buttonBorderRadius) var(--buttonBorderRadius) 0 0;
+  color: rgb(8,12,17);
```
[the remark this answers is not in the document]
Comment (dmcalester) [reply static/css/astro.core.css:1850]: @amalyah `astro.core.css`, `astro.css`, `astro.core.min.css` and `astro.min.css` are all compiled from the separate CSS files so will resolve with fixes above.

**Comment by dmcalester** [comment]
> @dmcalester Is there a desire or need to update the Readme and various links to reference select instead of drop-down?
> 
> Screen Shot 2021-02-15 at 9 22 57 AM


Yes please.

File: src/css/documentation/select-menu-README.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
-# Drop Down Menu
+# Select Menu
 
-Drop Down Menus allow users to select a value from a list of values.
+Select Menus allow users to select a value from a list of values.
 
 ## Guidelines
 
-- [Astro UXDS: Drop Down](https://www.astrouxds.com/ui-components/drop-down)
+- [Astro UXDS: Select](https://www.astrouxds.com/ui-components/drop-down)
```
Comment (dmcalester) [inline src/css/documentation/select-menu-README.md:7]: Please update URL to https://astrouxds.com/components/select/

**Comment by dmcalester** [comment]
**Chrome Fixes**
In Chrome on hover both the disabled and enabled state do a 1px shift on hover after initial load. Subsequent hover events are fine. In the disabled state the cursor flickers between the `not-allowed` and and `pointer` cursor. It appears the `not-allowed` occurs when hovering over the border of the `select` menu.

https://user-images.githubusercontent.com/60529018/108096836-49955380-7036-11eb-8ccb-bf7be9d83225.mov

**Safari/Firefox Fixes**
In both Firefox and Safari the disabled select state is accepting mouse enter state and changing the cursor to pointer and accepting click event swapping the background arrow. 

https://user-images.githubusercontent.com/60529018/108096201-8c0a6080-7035-11eb-8eea-b0586a9ae812.mov

- Please disable pointer cursor and/or use the `not-allowed` cursor.
- Please disable the arrow swap when clicking on the disabled select input

Also please target the `next` branch for merging. Typically planned fixed target `next` and hot fixes will target `master`. Thank you.

**Review by dmcalester — CHANGES_REQUESTED** [review]
Thanks Sasha,

Left a few comments for some visual/interaction fixes. Also please target the `next` branch.

File: src/css/common/_theme--dark.css [hunk 3]
```diff
@@ -343,6 +335,7 @@
 
   --inputBackgroundColor: var(--inputBackground);
   --inputBorderColor: var(--surfaceElements);
+  --inputBorderColorAlt: rgb(8, 12, 17);
```
[the remark this answers is not in the document]
Comment (Cheshire89) [reply src/css/common/_theme--dark.css:338]: this line is creating a color variable ... its just not referencing one as other variables do

File: src/css/common/_theme--dark.css [hunk 4]
```diff
@@ -17,6 +17,7 @@
 	--primaryDark: #3a87cf;
 	--primaryElementText: #080c11;
 	--inputBackground: #FFFFFF;
+  --inputDark: rgb(8,12,17);
```
Comment (amalyah) [inline src/css/common/_theme--dark.css:20]: Other color variables are using hex values. Could you do the same here for consistency?

File: src/css/common/_theme--dark.css [hunk 5]
```diff
@@ -343,6 +336,8 @@
 
   --inputBackgroundColor: var(--inputBackground);
   --inputBorderColor: var(--surfaceElements);
+  --inputBorderColorAlt: var(--inputDark);
+  --inputBorderColorDisabled: rgb(41, 42, 45);
```
Comment (amalyah) [inline src/css/common/_theme--dark.css:340]: Hex code please.

File: src/css/common/_theme--light.css [hunk 6]
```diff
@@ -17,6 +17,7 @@
   --primaryDarkHover: #0048724D; /* TODO: this is a temporary fix, the use of opacity from Sketch is new and not accounted for in CSS */
 	--primaryElementText: #ffffff;
 	--inputBackground: #ffffff;
+  --inputDark: rgb(8,12,17);
```
Comment (amalyah) [inline src/css/common/_theme--light.css:20]: Hex code please

File: src/css/common/_theme--light.css [hunk 7]
```diff
@@ -349,6 +344,8 @@
 
   --inputBackgroundColor: var(--inputBackground);
   --inputBorderColor: var(--primaryLight);
+  --inputBorderColorAlt: var(--inputDark);
+  --inputBorderColorDisabled: rgb(41, 42, 45);
```
Comment (amalyah) [inline src/css/common/_theme--light.css:348]: Hex code please

**Review by amalyah — CHANGES_REQUESTED** [review]
Few small comments for color consistency. If it's controlled by us in any way, the pop up menu appears off horizontally from the control when I don't think it was doing that before.

**Review by amalyah — APPROVED** [review]

**Review by cskerbo — APPROVED** [review]

File: static/css/astro.core.css [hunk 8]
```diff
@@ -1889,11 +1880,26 @@ label {
...
-	border-color: var(--selectActiveBorderColor);
-	outline-color: var(--selectActiveBorderColor);
+	outline: none;
+  border: 1px solid var(--primary);
+  border-radius: 3px;
+  border-radius: var(--buttonBorderRadius);
+}
+
+.rux-select:active:not(:disabled){
+  background-image: var(--select-active-caret);
```
Comment (dmcalester) [inline static/css/astro.core.css:1898]: Minor, but stylistically we use Hungarian notation for CSS Custom Properties to differentiate from class names.

**Review by dmcalester — CHANGES_REQUESTED** [review]
One minor code update and then a visual glitch in Chrome is present.

**Review by dmcalester — APPROVED** [review]
I’m still seeing the weird shift in Chrome, but only when refreshing the page and only when in the full storybook ui. When in an external it seems fine. Approving for now with a lien to keep an eye out for it.

**Pull request merged**